### ISSUE TITLE
Adding resources directory to framework causes carthage to fail

### ISSUE THREAD

**Issue by drekka** [Carthage/Carthage#1625]
* Carthage version: `0.18.1`: 
* Xcode: `8.2 beta (8C30a)`: 
* Are you using `--no-build`? - No
* Are you using `--no-use-binaries`? - Yes
* Are you using `--use-submodules`? - No

**Carthage Output**
```
KISS-Principle:unscriptdV2UI derekclarkson$ carthage update --no-use-binaries --platform iOS
*** Fetching unscriptdapi
*** Checking out unscriptdapi at "595663fa2ebc7f93dd8caa271d7285f90cdfb73e"
*** xcodebuild output can be found in /var/folders/q9/lt270bsn2j167slch5x83crh0000gn/T/carthage-xcodebuild.cSxVFI.log
*** Building scheme "Unscriptd iOS" in Unscriptd.xcodeproj
Failed to read file or folder at /Users/derekclarkson/Library/Developer/Xcode/DerivedData/Unscriptd-fjujrkqecvwvpbbotxtxrzhjuuzk/Build/Products/Release-iphoneos/Unscriptd.framework
```

I have a single dependency which is a Swift framework. The target has a copy files phases which copies files into the frameworks `/Resources` directory. Specifically it creates `Resources/en.lproj/Localizable.strings`. 

When this copy files phase is present the above error occurs. When I remove the phase then carthage works fine. I've checked, double checked and triple checked and the presence of files in the `Resources` directory seems to be the issue. 

Note that the actual building of the framework works just fine. It's something after that which fails. Perhaps code signing. The full carthage log does not indicate what the actual problem is.

**Comment by drekka** [comment]
I think I've found the problem. If I move the localisable strings from `Resources/en.lproj/Localizable.string` to `en.lproj/Localizable.string` (ie putting the localisable directory at the top level of the bundle) then everything seems to work.

Not sure if this is my bad or still represents a problem with carthage.

**Comment by mdiep** [comment]
Does the xcodebuild output give you any more information?

**Comment by drekka** [comment]
No. The xcode build log (either from cat'ing the log file or using --verbose) indicates a successful build of the framework file with no errors. It looks like carthage builds the framework then has a problem when it's trying to post process it. 

I first noticed this problem when I attempted to use the framework and got errors. The built framework file looked fine, but for some reason didn't want to work. Can't remember what the error I got in Xcode was.

**Comment by mdiep** [comment]
Is there a public project that exhibits the issue?

**Comment by drekka** [comment]
Sorry. No. Found this on a client project. Will put one together if I get the chance.

**Comment by 651893553** [comment]
I met the same problem, what's the solution?

**Comment by drekka** [comment]
I think I also came across this with the TwitterKit API. Twitter embeds all sorts of stuff in their framework files and I had to pull it apart and assemble a new framework to get it to work.

**Comment by mdiep** [comment]
If someone can link to a repo that demonstrates the issue, that will help to get it resolved.

**Comment by lottadot** [comment]
@drekka While it is true the `TwitterKit API` ships a fat framework ( https://github.com/CocoaPods/Specs/blob/master/Specs/b/a/9/TwitterKit/2.8.0/TwitterKit.podspec.json shows http": "https://kit-downloads.fabric.io/cocoapods/twitterkit/2.8.0/twitterkit.zip" ):

```
Cypher:carthage-1625 shane$ lipo -i /Users/shane/Projects/carthage-1625/Vendor/twitterkit/iOS/TwitterKit.framework/TwitterKit 
Architectures in the fat file: /Users/shane/Projects/carthage-1625/Vendor/twitterkit/iOS/TwitterKit.framework/TwitterKit are: armv7 armv7s i386 x86_64 arm64
```

I don't see how that fact pertains to this issue. The TwitterKit framework is only available via binary form (that I could find). There is no way to know whether, when it's built, there's a `copy files phase`. IMHO that doesn't help in trying to reproduce this Issue. Instead, what is needed is a public framework source that has a proper copy files phase.

**Comment by ikesyo** [comment]
Looks like you should have localization resources in a bundle then add the bundle to a framework's predefined `Copy Bundle Resources` phase as seen in [FormatterKit](https://github.com/mattt/FormatterKit).

**Comment by drekka** [comment]
@lottadot It's not about the copy files phase or how the framework is built. It's about the internal files and directories of the resulting framework causing Carthage to fail. Whether it's a framework I've built or one like TwitterKit, Carthage has issues with the internal structure of it. Specifically around the /Resources directory.

@ikesyo The TwitterKit framework includes it's localisation as a bundle and I had to extract it to get it to to work. But putting it in a bundle is something I might give a shot to see how it works. With mine I was trying to follow the Apple doco on framework files. See https://developer.apple.com/library/content/documentation/MacOSX/Conceptual/BPFrameworks/Frameworks.html which I've been using as a reference. 

I figure I must have the structure correct because I've bene able to get localised values out once I move the files inside the framework.
